File: Java_utils/mpeg2ts/StreamInfo.java
/**
 * 
 */
package com.ibm.mpeg2ts;



/**
 * @author evgenyh
 *
 */
public class StreamInfo {

	public enum TS_STREAM_TYPE{
		TS_STREAM_TYPE_MPEG2V,
		TS_STREAM_TYPE_H264,
		TS_STREAM_TYPE_AAC,
		TS_STREAM_TYPE_MP3,
		TS_STREAM_TYPE_AC3,
		TS_STREAM_TYPE_G711,	
		TS_STREAM_TYPE_UNKNOWN
	}

	public TS_STREAM_TYPE streamType;
	public int streamPID;
    
	@Override
	public String toString() {

		StringBuilder builder = new StringBuilder();
		
		builder.append("Type: "+streamType.name().substring(streamType.name().lastIndexOf("_")+1)+"\n");
		
		builder.append("PID: "+streamPID+"\n");
		
		return builder.toString();
		
	}
	
}

File: work/src/mpeg2ts/StreamInfo.go
package main

import "fmt"

const TS_STREAM_TYPE_MPEG2V = 0
const TS_STREAM_TYPE_H264 = 1
const TS_STREAM_TYPE_AAC = 2
const TS_STREAM_TYPE_MP3 = 3
const TS_STREAM_TYPE_AC3 = 4
const TS_STREAM_TYPE_G711 = 5
const TS_STREAM_TYPE_UNKNOWN = 6


type StreamInfo struct {
	streamType int
	streamPID int
}

//CONSTRUCTOR:
func NewStreamInfo(capacity int) *StreamInfo {
	return &StreamInfo{}
}

func ( s StreamInfo) ToString() string{
	 res := fmt.Sprintf( "Type: %d , PID: %d",s.streamType, s.streamPID)

	 return res
}
